Dev

dzhekzzzz opened the pull request:
changed many bugs
added random led timer generator
improved work stability

Antdigapony committed:
Merge branch 'main' into dev

Antdigapony committed:
Revision_2

dzhekzzzz committed:
Update LEDGAME.ino

dzhekzzzz committed:
Update LEDGAME.ino

Antdigapony commented on LEDGAME.ino:
```diff
@@ -3,7 +3,6 @@
 int a = 1;
```
change this variable name from 'a' to 'gameState' or 'gameStatus'

Antdigapony replied:
also I recommend to use the directives to the pre-processor:

#define GAME_SET_UP (1)
#define GAME_START (2)
#define GAME_RUNNING (3)
#define GAME_END (4)

and set the former 'a' variable gameState:

int gameState = GAME_SET_UP;

Antdigapony commented on LEDGAME.ino:
```diff
@@ -35,30 +33,26 @@ class Timer
 Timer ledTimer;
 
+
 bool isLedTimerRun()
```
send this func to the class Timer
or use Timer-class function instead this func

Antdigapony commented on LEDGAME.ino:
```diff
@@ -149,47 +125,60 @@ int buttWait(int num)
+        random_led(b); //turn on random led
+        ledTimer.setTimer(t); //setting time for leds
+        timer.i_oldLedTimer = millis();
+        while ( isLedTimerRun() == true )
```
recommendations: something like that:

while(ledTimer.isTimerRunning())
{
   if (buttonIsPressed()) 
   {
       break;
   }
}

Antdigapony commented on LEDGAME.ino:
```diff
@@ -149,47 +125,60 @@ int buttWait(int num)
         buttWait(b);
-        clearbutt(); //очищаю значение кнопок  
-        led_off(); //выключаем все лампочки
+        clearbutt(); //clearing butt bufer
```
it's better to send this func with scores to the GAME_OVER case or to the separate state/case SHOW_SCORES

dzhekzzzz committed:
Update LEDGAME.ino

dzhekzzzz committed:
Update LEDGAME.ino

Antdigapony commented on LEDGAME.ino:
```diff
@@ -1,12 +1,14 @@
 int buttInit[4] = {4, 7, 10, 12} ;
 int ledInit[4] = {3, 9, 5, 2} ;
 int butt[5];
+#define GAME_SET_N_START 1
```
just a small remark - use () - round brackets to avoid mistakes
#define GAME_SET_IN_START (1)

Antdigapony commented on LEDGAME.ino:
```diff
@@ -20,127 +22,95 @@ void setup()
 {
   public:
   unsigned long  i_timer;
   unsigned long  i_ledTimer;
```
this class-field or class-variable is not used anywhere - maybe we should delete it?

Antdigapony commented on LEDGAME.ino:
```diff
@@ -20,127 +22,95 @@ void setup()
   unsigned long  i_ledTimer;
   unsigned long  i_oldLedTimer;
   unsigned long  limitValue;
-  unsigned long old_i_timer;
```
and this too -

Antdigapony commented on LEDGAME.ino:
```diff
@@ -20,127 +22,95 @@ void setup()
 class Timer
 {
   public:
   unsigned long  i_timer;
```
it's better to use ONE STYLE for the variables:
camelCase without _ lowerspaces
and I propose that variables should also "tell" something about themselves:

unsigned long currentTime;
unsigned long previousTime;

Antdigapony commented on LEDGAME.ino:
```diff
@@ -20,127 +22,95 @@ void setup()
   unsigned long  limitValue;
-  unsigned long old_i_timer;
+  unsigned long  old_i_timer;
+  unsigned long  newTimeLed;
```
this variable is also un-used

Antdigapony commented on LEDGAME.ino:
```diff
@@ -20,127 +22,95 @@ void setup()
 }
 
+ //turn on random led
+int randomLed(int num)
```
better call it
randomLedOn - cause we are turning the random led on

Antdigapony commented on LEDGAME.ino:
```diff
@@ -20,127 +22,95 @@ void setup()
   }  
 
 //выдаем результат в зависимости от нажатой кнопки 
 int buttWait(int num)
```
proposal to delete this function
it is redundant

Antdigapony replied:
instead - create a function:

bool checkButtonPressed(int num)
{
    return (digitalRead(buttInit[num]) == HIGH)
}

Antdigapony replied:
you can launch this function in a cycle and use for the ALL buttons check one by one

Antdigapony commented on LEDGAME.ino:
```diff
@@ -149,52 +119,74 @@ int buttWait(int num)
-{
-  while (butt[0] == 0) 
+ //waiting for start button
+void startButtonWait()
```
now you can use:
while (!checkButtonPressed(0))

Antdigapony commented on LEDGAME.ino:
```diff
@@ -149,52 +119,74 @@ int buttWait(int num)
+        while ( ledTimer.isTimerRunning() == true )
         {
-        a = 3;
+          if (digitalRead(buttInit[b]) == HIGH) 
```
you can use checkButtonPressed here too

Antdigapony commented on LEDGAME.ino:
```diff
@@ -149,52 +119,74 @@ int buttWait(int num)
         Serial.print("\n \n");
-        delay(500);
-
+        waitTimer.setTimer(t);
```
waitTimer - were do you set this Timer and were do you update this Timer ?

dzhekzzzz replied:
setting timer by parameter t, function updating itself in func "waitTimer.isTimerRunning()" below

Antdigapony commented on LEDGAME.ino:
```diff
@@ -149,52 +119,74 @@ int buttWait(int num)
-        timer.isTimerRun();
-        random_led(b);
-        ledTimer.setTimer(500); //ставим сколько дает времени на нажатие
         buttWait(b);
```
this is redundant or can be written here - to compare the button and buttonInit

dzhekzzzz committed:
Update LEDGAME.ino

dzhekzzzz committed:
Update LEDGAME.ino

Antdigapony commented on LEDGAME.ino:
```diff
@@ -1,12 +1,15 @@
 int buttInit[4] = {4, 7, 10, 12} ;
 int ledInit[4] = {3, 9, 5, 2} ;
-int butt[5];
+int butt[4];
```
it's better to rename this array to buttState, cause it keeps the state of the buttons and it's a good style to init the array with 0s or 1s

Antdigapony commented on LEDGAME.ino:
```diff
@@ -1,12 +1,15 @@
+int butt[4];
+#define GAME_SET_N_START (1)
+#define GAME_RUNNING (2)
+#define GAME_END (3)
```
please, add separate state of the GAME_SCORES where you will show the player's scores for 3-5 seconds

Antdigapony commented on LEDGAME.ino:
```diff
@@ -1,12 +1,15 @@
+#define GAME_RUNNING (2)
+#define GAME_END (3)
 
 void setup() 
```
please, move the setup() to the loop(), so in the end, there must be:
- your defines and variables
- your classes and funcs
- setup()
- loop()

Antdigapony commented on LEDGAME.ino:
```diff
@@ -20,181 +23,187 @@ void setup()
+      {
+        newTimeValue = millis();
+        
+        if ( (newTimeValue - oldTimeValue) <= limitValue)
```
you can simply make the same with:
return ( (newTimeValue - oldTimeValue) <= limitValue )

dzhekzzzz replied:
I need to return the value of true \ false, if I change it will this func return it to me?

Antdigapony replied:
yes, it will return the result of the condition's checking

Antdigapony commented on LEDGAME.ino:
```diff
@@ -20,181 +23,187 @@ void setup()
 }
 
+ //turn on random led
+int randomLedOn(int num)
```
un-correct function's name - this function turns the Led ON - not the random LED, but any LED with number num

Antdigapony commented on LEDGAME.ino:
```diff
@@ -20,181 +23,187 @@ void setup()
-//выдаем результат в зависимости от нажатой кнопки 
-int buttWait(int num)
+//буфер с нажатыми кнопками
+int WhichButtPressed()
```
please, use one style:
- camelCase
- and start the function names with a lower-case letter

Antdigapony replied:
un-correct name of the function: this function does not return anything and it just sets but[c] = c; if any button is pressed.
Maybe it is better to call it checkAnyButtonPressed() ???
and think about the buttonOldState and buttonNewState

Antdigapony commented on LEDGAME.ino:
```diff
@@ -20,181 +23,187 @@ void setup()
+  
+  for (int c = 1; c < 4; c++)
+  {
+   if (digitalRead(buttInit[c]) == HIGH)
```
you are using both arrays:
- buttonInit[]
- butt

Maybe it is more convenient to change their names to :
- newButtonState[]
- oldButtonState[]

dzhekzzzz replied:
first arr is only for initing butts, i think its better to call buttInin
second arr is for buffer with butt states, i rename it in buttState

i think these names are better

Antdigapony commented on LEDGAME.ino:
```diff
@@ -20,181 +23,187 @@ void setup()
     {
         butt[0] = 1; 
-      for (int i = 1; i < 5; i++)
+      for (int i = 1; i < 4; i++)
```
you can avoid the 'magic number' issues and use a #define NUM_BUTTONS or BUTTONS_QUANTITY

Antdigapony commented on LEDGAME.ino:
```diff
@@ -20,181 +23,187 @@ void setup()
+   }
 }
+
+void PrintScore()
```
styling - use lower case letter for the functions, please

Antdigapony commented on LEDGAME.ino:
```diff
@@ -20,181 +23,187 @@ void setup()
+
 
-        return 0;
+int Logic(int num)
```
bad name for the function - Logic does not tell anything what the function does.

maybe call it: checkResult() or checkPressing(). something like that, which will tell what is happening here - in this function

Antdigapony commented on LEDGAME.ino:
```diff
@@ -20,181 +23,187 @@ void setup()
-        return 0;
+int Logic(int num)
+{
+ if (butt[num] == num && i == 1)
```
if you are using a global variable as a flag - i
it's better to call what this flag-variable determines

Antdigapony commented on LEDGAME.ino:
```diff
@@ -20,181 +23,187 @@ void setup()
-        timer.setTimer(15000); //сколько будет идти игра
-        timer.updateTimer(); // включаем таймер
-        a = 2;
+        startButtonWait(); //waiting for start button
```
what will happen - if the player will not press the START_BUTTON here ?
will the system react on this ?

dzhekzzzz replied:
system will wait for a start_button press for a decades

dzhekzzzz commented:
changed many bugs
improved work stability
added new cool features
added random time delay between leds blimp
slightly changed design of the information displayed on the screen

for modders: redesigned code syntax (may be easier to read, correct markup and function names)
